thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -95,16 +103,30 @@ public void run() {
           if (isAllDatasetUpdated(detectionConfig, detection2DatasetMap.get(detectionConfig), datasetConfigMap)) {
             if (isWithinSchedulingWindow(detection2DatasetMap.get(detectionConfig), datasetConfigMap)) {
               //TODO: additional check is required if detection is based on aggregated value across multiple data points
-              createDetectionTask(detectionConfig);
+              long endtime = System.currentTimeMillis();
+              createDetectionTask(detectionConfig, endtime);
+              detectionIdToWatermarkMap.put(detectionConfig.getId(), endtime);
               ThirdeyeMetricsUtil.eventScheduledTaskCounter.inc();
               taskCount++;
             } else {
               LOG.warn("Unable to schedule a task for {}, because it is out of scheduling window.", detectionConfigId);
             }
           }
+
+          // Note: Fallback SLA & Data availability SLA are independent of each other.
+          // For example, if an event doesn't arrive within 24 hours, do a fallback.
+          // On the other hand, a user can setup an SLA alert if there is no data for 3 days.
           if (needFallback(detectionConfig)) {
             LOG.info("Scheduling a task for detection {} due to the fallback mechanism.", detectionConfigId);
-            createDetectionTask(detectionConfig);
+            long endtime = System.currentTimeMillis();
+            createDetectionTask(detectionConfig, endtime);
+
+            if (DetectionUtils.isDataAvailabilityCheckEnabled(detectionConfig)) {
+              createDataAvailabilityTask(detectionConfig, endtime);
```

Comment:
Do you create data availability task elsewhere? Is this the only place to trigger data availability task?

Reply:
Discussed offline. Data SLA task will be created in 2 cases.
a. Whenever the DetectionCronScheduler schedules a detection task.
b. Whenever the DataAvailabilityTaskScheduler does a fallback.
Note that we do not need to schedule a Data SLA task when a Data availability event arrives.